### PR TITLE
chore: develop getDatePickerDefaultizedProps function

### PR DESCRIPTION
**Pull request by XaviAlonsoo**

### CODE REVIEW

File: libs/pickers/src/internals/hooks/validation/useValidation.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export type Validator<TValue, TDate, TError, TValidationProps> = (params: {
+  utils: PickersAdapter<TDate>;
+  value: TValue;
+  props: Omit<TValidationProps, 'value' | 'onError'>;
+}) => TError;
+
+export function useValidation<TValue, TDate, TError, TValidationProps extends {}>(
+  props: ValidationProps<TError, TValue, TValidationProps>,
+  validate: Validator<TValue, TDate, TError, TValidationProps>,
+  isSameError: (a: TError, b: TError | null) => boolean,
```
[the remark this answers is not in the document]
Comment (XaviAlonsoo) [reply libs/pickers/src/internals/hooks/validation/useValidation.ts:33]: to not call onError callback given by user, twice on two referentially identical errors.

File: libs/pickers/src/LocalizationProvider/index.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
 export { LocalizationProvider, PickersAdapterContext } from './LocalizationProvider';
-export type { LocalizationProviderProps, PickersAdapterContextValue } from './LocalizationProvider';
+export type { LocalizationProviderProps, PickersAdapter, PickersAdapterContextValue } from './LocalizationProvider';
```
Comment (XaviAlonsoo) [inline libs/pickers/src/LocalizationProvider/index.ts:2]: export * from './LocalizationProvider'

File: libs/pickers/src/internals/hooks/validation/useDateValidation.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  return React.useCallback(
+    (day: TDate | null) =>
+      validateDate({ utils, props: { disablePast, disableFuture, minDate, maxDate }, value: day }) !== null,
+    [utils, disablePast, disableFuture, minDate, maxDate],
+  );
+};
+
+export const isSameDateError = (a: DateValidationError, b: DateValidationError) => a === b;
+
+export const useDateValidation = <TDate>(
```
Comment (XaviAlonsoo) [inline libs/pickers/src/internals/hooks/validation/useDateValidation.ts:57]: remove

File: libs/pickers/src/internals/hooks/validation/useDateValidation.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+export type DateValidationError = CommonDateTimeValidationError | 'minDate' | 'maxDate';
+
+export const validateDate: Validator<any | null, any, DateValidationError, BaseDateValidationProps<any>> = ({
+  props,
+  value,
+  utils,
+}): DateValidationError => {
+  if (value === null) return value;
+
+  const now = utils.date()!;
```
Comment (mdeljavan) [inline libs/pickers/src/internals/hooks/validation/useDateValidation.ts:17]: if you are sure that the `util.date()` will always return a non-null value, I think you should change its type definition.

**Review by mdeljavan — APPROVED** [review]

**Pull request merged**